pallets/connectors-gateway/connectors-gateway-routers/src/ethereum_xcm.rs

```diff
@@ -0,0 +1,256 @@
+// Copyright 2021 Centrifuge Foundation (centrifuge.io).
+//
+// This file is part of the Centrifuge chain project.
+// Centrifuge is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version (see http://www.gnu.org/licenses).
+// Centrifuge is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+use core::convert::TryFrom;
+
+use cfg_traits::connectors::Codec;
+use codec::{Decode, Encode, MaxEncodedLen};
+use ethabi::{Bytes, Contract};
+use frame_support::{
+	dispatch::DispatchResult, sp_runtime::DispatchError, traits::OriginTrait, weights::Weight,
+};
+use pallet_xcm_transactor::{Currency, CurrencyPayment, TransactWeights};
+use scale_info::TypeInfo;
+use sp_core::{bounded::BoundedVec, ConstU32, H160, U256};
+use sp_std::{boxed::Box, marker::PhantomData, vec, vec::Vec};
+use xcm::{
+	v2::{MultiLocation, OriginKind},
+	VersionedMultiLocation,
+};
+
+use crate::{AccountIdOf, CurrencyIdOf, MessageOf};
+
+#[derive(Debug, Encode, Decode, Clone, PartialEq, Eq, TypeInfo, MaxEncodedLen)]
+pub struct EthereumXCMRouter<T>
+where
+	T: frame_system::Config + pallet_xcm_transactor::Config + pallet_connectors_gateway::Config,
+{
+	pub xcm_domain: XcmDomain<CurrencyIdOf<T>>,
+	pub _marker: PhantomData<T>,
+}
+
+impl<T> EthereumXCMRouter<T>
+where
+	T: frame_system::Config + pallet_xcm_transactor::Config + pallet_connectors_gateway::Config,
+{
+	pub fn do_init(&self) -> DispatchResult {
+		pallet_xcm_transactor::Pallet::<T>::set_transact_info(
```

Comment:
The idea was to require a governance proposal in order to submit and thus be able to use a router, for the sake of transparency. I don't necessarily oppose this approach tho, but it could happen that by the time this is published we might need to change some of the values through a proposal anyways. I think it's fine to keep it, unless if someone has different opinions 👍

Reply:
Adding a router will also be done via a governance proposal, this logic will be triggered when the router is added. Just wanted to mention that.